### DISCUSSION TITLE
WARNING: some attributes cannot be read from corsair-cpro kernel driver

### DISCUSSION THREAD

**Discussion by Tech127x** [liquidctl/liquidctl#711] [Q&A] [answered]
Is this normal?  Did I not do something on install/config/whatever?  Thanks!

**Comment by aleksamagicka** [comment, marked as answer]
Yes, it's normal. During initialization, liquidctl will notice that you have the corsair-cpro kernel driver loaded and will request some info from it, instead of contacting the device directly. However, the kernel driver did not expose firmware and bootloader versions until [very recently](https://git.kernel.org/pub/scm/linux/kernel/git/groeck/linux-staging.git/commit/drivers/hwmon/corsair-cpro.c?h=hwmon-next&id=5997eb60f896830126bc1783465b678b110e2fdd), so liquidctl doesn't ask it for that right now and logs the message you get instead.
